DatWeatherDoe/API/WeatherRepository.swift

```diff
@@ -61,9 +54,8 @@ final class WeatherRepository: WeatherRepositoryType {
 
         URLSession.shared.dataTask(with: url) { [weak self] data, _, error in
             guard let `self` = self, let data = data, error == nil else {
```

Comment:
this results in a compilation error for me:
`Variable declared in 'guard' condition is not usable in its body`

using something like this works:

```
guard let strongSelf = self else {
    self?.logger...
  
    strongSelf.parseResponse...
}
```